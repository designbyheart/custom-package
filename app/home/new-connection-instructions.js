// @flow

import React, { PureComponent } from 'react'
import {
  ImageBackground,
  Image,
  TouchableOpacity,
  StyleSheet,
  Linking,
} from 'react-native'
import Snackbar from 'react-native-snackbar'

import { Container, CustomView, CustomText, CustomButton } from '../components'
import {
  OFFSET_3X,
  OFFSET_2X,
  OFFSET_1X,
  cmGrey1,
  white,
  verticalBreakpoint,
  venetianRed,
} from '../common/styles'

const { extraSmall, small, medium, large } = verticalBreakpoint

export class NewConnectionInstructions extends PureComponent<{}> {
  render() {
    return (
      <ImageBackground
        source={require('../images/connection-items-placeholder.png')}
        style={styles.backgroundImage}
      >
        <Container center style={[styles.contentContainer]}>
          <CustomView center>
            <Image
              source={require('../images/CredentialGraphic.png')}
              resizeMode="contain"
              style={
                extraSmall ? styles.credentialSmallCard : styles.emptyStyle
              }
            />
            <CustomText transparentBg size="h3b" bold style={[styles.headline]}>
              You now have a digital wallet!
            </CustomText>
            <CustomText transparentBg bold size="h5" style={[styles.title]}>
              Want to see how it works?
            </CustomText>
            <CustomText transparentBg size="h6" style={[styles.text]}>
              We have setup an optional tutorial site for you to go through
              using this Connect.Me app. To start this process, go to{' '}
              <CustomText
                transparentBg
                size="h6"
                style={[styles.link]}
                onPress={this.openTryConnectMe}
              >
                try.connect.me
              </CustomText>{' '}
              in a desktop browser, or click the button below!
            </CustomText>
            <CustomView center style={[styles.buttonContainer]}>
              <CustomButton
                title="Start Tutorial"
                style={[styles.startTutorialButton]}
                customColor={buttonColor}
                onPress={this.openTryConnectMe}
              />
            </CustomView>
          </CustomView>
        </Container>
      </ImageBackground>
    )
  }

  openTryConnectMe = () => {
    Linking.openURL('https://try.connect.me').catch(() => {
      Snackbar.show({
        title:
          'Could not open Tutorial website. No capable browser found to open it.',
        duration: Snackbar.LENGTH_LONG,
        backgroundColor: venetianRed,
        color: white,
      })
    })
  }
}

const buttonColor = {
  color: white,
  fontWeight: '600',
  fontSize: 18,
}

// we just want an approx height because we want to push images and empty view
// just a little bit down, and on smaller devices as well it would be
// help in jamming down view
const approxHeaderHeight = 64
const verticalSpacing1X = small ? OFFSET_1X / 2 : OFFSET_1X

const styles = StyleSheet.create({
  backgroundImage: {
    width: '100%',
    height: '100%',
  },
  contentContainer: {
    marginTop: small ? approxHeaderHeight : 0,
  },
  credentialCard: {
    marginBottom: verticalSpacing1X,
  },
  credentialSmallCard: {
    height: 90,
  },
  headline: {
    color: '#EB9B2D',
    marginVertical: verticalSpacing1X,
    textAlign: 'center',
  },
  title: {
    marginVertical: verticalSpacing1X,
    color: cmGrey1,
  },
  text: {
    marginVertical: verticalSpacing1X,
    marginHorizontal: '5%',
    color: cmGrey1,
    textAlign: 'center',
  },
  link: {
    color: '#EB9B2D',
  },
  buttonContainer: {
    marginTop: small ? '2%' : medium ? '5%' : '10%',
  },
  startTutorialButton: {
    borderRadius: 5,
    backgroundColor: '#86B93B',
    marginHorizontal: '6%',
  },
  // creating empty style so that Image component does not re-render
  // every time, because we would be passing "{}", and hence a new object
  // would be created every time, and hence equality operator
  // would see it as different prop and hence it would render again
  // by using this emptyStyle we are making sure that it does re-render
  // just because of new object every time due to styles being empty
  emptyStyle: {},
})
