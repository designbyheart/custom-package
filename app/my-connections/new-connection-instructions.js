// @flow

import React, { PureComponent } from 'react'
import {
  ImageBackground,
  Image,
  TouchableOpacity,
  StyleSheet,
  Linking,
} from 'react-native'
import Snackbar from 'react-native-snackbar'

import { Container, CustomView, CustomText, CustomButton } from '../components'
import {
  OFFSET_3X,
  OFFSET_2X,
  OFFSET_1X,
  cmGrey1,
  white,
  verticalBreakpoint,
  venetianRed,
} from '../common/styles'

import type { NewConnectionInstructionsProps } from './type-my-connections'

const { extraSmall, small, medium, large } = verticalBreakpoint

export class NewConnectionInstructions extends PureComponent<
  NewConnectionInstructionsProps,
  void
> {
  render() {
    const { usingProductionNetwork } = this.props
    return (
      <ImageBackground
        source={require('../images/connection-items-placeholder.png')}
        style={styles.backgroundImage}
      >
        <Container center style={[styles.contentContainer]}>
          <CustomView center>
            <Image
              source={require('../images/CredentialGraphic.png')}
              resizeMode="contain"
              style={
                extraSmall || small || medium
                  ? styles.credentialSmallCard
                  : styles.emptyStyle
              }
            />
            <CustomText transparentBg size="h3b" bold style={[styles.headline]}>
              You now have a digital wallet!
            </CustomText>
            <CustomText transparentBg bold size="h5" style={[styles.title]}>
              Want to see how it works?
            </CustomText>
            {usingProductionNetwork && (
              <CustomText transparentBg size="h6" style={[styles.text]}>
                We have setup an optional tutorial site for you to go through
                using this Connect.Me app. To start this process, go to
                try.connect.me in a desktop browser and click Start Tutorial.
              </CustomText>
            )}
            {!usingProductionNetwork && (
              <CustomText transparentBg size="h6" style={[styles.text]}>
                We see you are not on the live network. Get with an Evernym team
                member to help you use Connect.Me!
              </CustomText>
            )}
          </CustomView>
        </Container>
      </ImageBackground>
    )
  }

  openTryConnectMe = () => {
    Linking.openURL('https://try.connect.me').catch(() => {
      Snackbar.show({
        title:
          'Could not open Tutorial website. No capable browser found to open it.',
        duration: Snackbar.LENGTH_LONG,
        backgroundColor: venetianRed,
        color: white,
      })
    })
  }
}

const buttonColor = {
  color: white,
  fontWeight: '600',
  fontSize: 18,
}

// we just want an approx height because we want to push images and empty view
// just a little bit down, and on smaller devices as well it would be
// help in jamming down view
const approxHeaderHeight = 64
const verticalSpacing1X = small ? OFFSET_1X / 2 : OFFSET_1X

const styles = StyleSheet.create({
  backgroundImage: {
    width: '100%',
    height: '100%',
  },
  contentContainer: {
    marginTop: small ? approxHeaderHeight : 0,
  },
  credentialCard: {
    marginBottom: verticalSpacing1X,
  },
  credentialSmallCard: {
    height: 90,
  },
  headline: {
    color: '#EB9B2D',
    marginVertical: verticalSpacing1X,
    textAlign: 'center',
  },
  title: {
    marginVertical: verticalSpacing1X,
    color: cmGrey1,
  },
  text: {
    marginVertical: verticalSpacing1X,
    marginHorizontal: '5%',
    color: cmGrey1,
    textAlign: 'center',
  },
  link: {
    color: '#EB9B2D',
  },
  buttonContainer: {
    marginTop: small ? '2%' : medium ? '5%' : '10%',
  },
  startTutorialButton: {
    borderRadius: 5,
    backgroundColor: '#86B93B',
    marginHorizontal: '6%',
  },
  // creating empty style so that Image component does not re-render
  // every time, because we would be passing "{}", and hence a new object
  // would be created every time, and hence equality operator
  // would see it as different prop and hence it would render again
  // by using this emptyStyle we are making sure that it does re-render
  // just because of new object every time due to styles being empty
  emptyStyle: {},
})
