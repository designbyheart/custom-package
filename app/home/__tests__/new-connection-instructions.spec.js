// @flow

import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'
import { NewConnectionInstructions } from '../new-connection-instructions'

describe('<NewConnectionInstructions />', () => {
  it('should match snapshot', () => {
    const component = renderer.create(<NewConnectionInstructions />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
