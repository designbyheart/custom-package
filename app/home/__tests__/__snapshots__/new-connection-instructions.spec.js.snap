// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewConnectionInstructions /> should match snapshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    center={true}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginTop": 64,
      }
    }
  >
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "height": 90,
          }
        }
      />
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontSize": 23,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            Object {
              "fontWeight": "bold",
            },
            null,
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "color": "#EB9B2D",
              "marginVertical": 5,
              "textAlign": "center",
            },
          ]
        }
      >
        You now have a digital wallet!
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            Object {
              "fontWeight": "bold",
            },
            null,
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "color": "#505050",
              "marginVertical": 5,
            },
          ]
        }
      >
        Want to see how it works?
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontSize": 15,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            null,
            null,
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "color": "#505050",
              "marginHorizontal": "5%",
              "marginVertical": 5,
              "textAlign": "center",
            },
          ]
        }
      >
        We have setup an optional tutorial site for you to go through using this Connect.Me app. To start this process, go to
         
        <Text
          allowFontScaling={false}
          onPress={[Function]}
          style={
            Array [
              Object {
                "fontSize": 15,
              },
              Object {
                "fontFamily": "Lato",
              },
              Object {
                "backgroundColor": "#333333",
                "color": "#ffffff",
              },
              null,
              null,
              Object {
                "backgroundColor": "transparent",
              },
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              Object {
                "color": "#EB9B2D",
              },
            ]
          }
        >
          try.connect.me
        </Text>
         
        in a desktop browser, or click the button below!
      </Text>
      <View
        center={true}
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginTop": "2%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "marginLeft": 15,
                "marginRight": 15,
              },
              undefined,
              Object {
                "marginLeft": 0,
                "marginRight": 0,
              },
              undefined,
            ]
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "#86B93B",
                  "borderRadius": 5,
                  "marginHorizontal": "6%",
                },
              ]
            }
          >
            <View
              pointerEvents="box-only"
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#9E9E9E",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "padding": 19,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Object {
                    "backgroundColor": "#ffffff",
                  },
                  undefined,
                  Object {
                    "padding": 12,
                  },
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    null,
                    Object {
                      "backgroundColor": "#86B93B",
                      "borderRadius": 5,
                      "marginHorizontal": "6%",
                    },
                  ],
                  undefined,
                  undefined,
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {},
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Array [
                      Object {
                        "color": "white",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#ffffff",
                      },
                      Object {
                        "fontSize": 14,
                      },
                      Object {
                        "fontSize": 18,
                      },
                      undefined,
                      Object {
                        "fontWeight": "600",
                      },
                      undefined,
                      undefined,
                    ],
                  ]
                }
              >
                Start Tutorial
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`<NewConnectionInstructions /> should match snapshot for test network 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    center={true}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginTop": 64,
      }
    }
  >
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "height": 90,
          }
        }
      />
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontSize": 23,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            Object {
              "fontWeight": "bold",
            },
            null,
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "color": "#EB9B2D",
              "marginVertical": 5,
              "textAlign": "center",
            },
          ]
        }
      >
        You now have a digital wallet!
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            Object {
              "fontWeight": "bold",
            },
            null,
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "color": "#505050",
              "marginVertical": 5,
            },
          ]
        }
      >
        Want to see how it works?
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontSize": 15,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            null,
            null,
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "color": "#505050",
              "marginHorizontal": "5%",
              "marginVertical": 5,
              "textAlign": "center",
            },
          ]
        }
      >
        We see you are not on the live network. Get with an Evernym team member to help you use Connect.Me!
      </Text>
    </View>
  </View>
</View>
`;
